### ISSUE TITLE
Table header style issues

### ISSUE THREAD

**Issue by ciphersimian** [StarCitizenTools/mediawiki-skins-Citizen#902]
## Describe the Bug
First of all I'm not entirely sure this is a "bug" persay, and some of this may be subjective, but other parts of it definitely seem broken so just a disclaimer that I understand some of this may be "working as intended" so consider those parts feature requests rather than bugs. :)

Generally I'm seeing some major differences in tables across my wiki after swapping from Vector 2022 to Citizen and some of these aren't expected.

1. Create a new page with the following content
```
<noinclude>
{| class="wikitable sortable" style="text-align: center;" border="1"
</noinclude>
! rowspan="2" | Level !! rowspan="2" | Name !! rowspan="1" colspan="3" | Max% !! colspan="8" | Launcher !! colspan="5" | Ammo
|-
! rowspan="1" style="color: red" | DPS !! rowspan="1" style="color: red" | EPS !! rowspan="1" style="color: red" | MPAS !! Max% !! style="color: red" | Reload !! Rounds !! Energy !! Race !! Class !! Buffs !! Negs !! rowspan="1" | Max% !! rowspan="1" style="color: red" | Damage !! rowspan="1" | Type(s) !! rowspan="1" | Range !! rowspan="1" | Stack
<noinclude>
|}
</noinclude>
```
2. This content represents a multi-row table header ONLY, it's a template that is intended to be combined with rows on various different context-sensitive pages.
3. Compare this table header in the Vector (2022) and Citizen skins with URLs like:
- https://your/wiki/index.php?title=Template:Launchers_Header&useskin=vector-2022
- https://your/wiki/index.php?title=Template:Launchers_Header&useskin=citizen
4. For me this yields (Vector (2022) top, Citizen bottom, combined into the same image to maintain scale):
![image](https://github.com/StarCitizenTools/mediawiki-skins-Citizen/assets/38382607/eb3ab8c0-bc4d-426c-8ee3-492a753e3500)

Also for the record, my preferences:
![image](https://github.com/StarCitizenTools/mediawiki-skins-Citizen/assets/38382607/13f2c67a-ebe3-4593-b6d8-cc336c34937d)

## Expected Behavior
In general I expect similar results so that I can create content that works equally well for my users who prefer one skin vs. the other.

The major discrepancies I note here are:

- Table headers are not centered or bolded in Citizen and these are primary functions of the `!` table header specifier in my opinion (https://www.mediawiki.org/wiki/Help:Tables#Table_headers, "Headers usually show up bold and centered by default.", but perhaps this skin is where "usually" ends.  :)
- The mediawiki `border="1"` option on tables doesn't seem to be handled well as it's causing the style color for the text to be applied to the cell border in this case. It is also messing up the rounding of corners of the header rows. This is visible in both the lower left corner of the "Level" cell as well as the lower left corner of the "DPS" cell. It seems likely this is related to the combination of a multi-row header, `rowspan=2` of the "Name" cell, and `border="1"`. Alternatively it could also be the lower right corner of the "Stack" cell rather than the lower left "Level" cell depending on whether header rows are intended to be visually separated by rounded bottom corners or not (e.g. the table I mention below is like that, though other tables on my wiki aren't, so I'm not sure yet what accounts for that and what the intention is).
- The cell padding is significantly higher in Citizen to the point where larger tables like this end up significantly wider; here the difference is ~245 pixels which is hard to account for. Some difference is understandable and tolerable, but I think it's currently a little too high. A good example of this is a large table like https://starcitizen.tools/List_of_pledge_vehicles

To the extent that these things are intentional and aren't going to change, it would be helpful if you could provide guidance on the best way to modify them globally for my instance, perhaps by adding tips for these specific items to https://github.com/StarCitizenTools/mediawiki-skins-Citizen/wiki/Customizing-Citizen-styles and ensuring these things are easy to customize if they aren't already?

## System
 - Citizen version `v2.21.0`
 - MediaWiki version `v1.42.1`
 - Server OS: `Unknown`
 - Installed extensions and versions
   - [DynamicPageList3](https://www.mediawiki.org/wiki/Extension:DynamicPageList3) `v3.5.2`
   - [ParserFunctions](https://www.mediawiki.org/wiki/Extension:ParserFunctions) `v1.6.1`
   - [Scribunto](https://www.mediawiki.org/wiki/Extension:Scribunto)
   - [TemplateStyles](https://www.mediawiki.org/wiki/Extension:TemplateStyles) `v1.0 (7bc4f37) 00:36, 4 July 2024`
   - [Disambiguator](https://www.mediawiki.org/wiki/Extension:Disambiguator) `v1.4 (00030e8) 00:43, 27 June 2024`
   - [IPBLoginAuth](https://www.mediawiki.org/wiki/Extension:IPBAuthLogin) `v1.0.1`
 - PHP version `8.3.8 (cgi-fcgi)`

**Comment by alistair3149** [comment]
Thanks for the detailed report. Some of the issues that you raised are bugs, and some of them are deliberate choices.

> Table headers are not centered or bolded in Citizen...

* The font weight is unintentional and the bold behavior is restored with 232fd0cc4b8ae517c6eba75f218bababc77535ee.
* For the center alignment, it is intended to be start (left on LTR language) aligned due to readability issue. For example, when the `<th>` element is center-aligned and wider than the viewport or container, it won't be visible until the table is scrolled. If you wish to re-enable this behavior, you can either put `style="text-align: center"` next to the `!`, or apply it wiki-wide through `MediaWiki:Citizen.css` with the following CSS:
```CSS
.wikitable th {
  text-align: center;
}
 ```

> The mediawiki border="1" option on tables doesn't seem to be handled well...

It is unintentional and should be patched with 442c272cabc3ffc0803eac8fcf5040910d2ad7e0.

> The cell padding is significantly higher in Citizen...

It is intentional and Citizen already has a more compact padding compared to many other sites. The example on Star Citizen Wiki is not a wikitable, but a datatable provided by extensions like Semantic MediaWiki and Cargo. If you wish to adjust the padding, you can adjust individually like this: https://www.mediawiki.org/wiki/Help:Tables#Padding, or apply it wiki-wide through `MediaWiki:Citizen.css` with the following CSS:
```CSS
.wikitable th {
  padding: ...;
}
 ```

**Comment by alistair3149** [comment]
> In an effort to explore what is going on here and try and find some workarounds I discovered that Citizen is not honoring styles on header rows at all, so maybe some of these issues are the result of a bug?
> 
> Take this simplified example:
> 
> ```
> {| class="wikitable"
> |- style="border: 5px solid red;"
> ! Level !! Name
> |-
> ! DPS !! EPS
> |}
> ```
> 
> With Vector (2022) this yields: ![image](https://private-user-images.githubusercontent.com/38382607/346315634-00e9b613-e2ae-486b-9a43-315993acd8a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzMzQxMDEsIm5iZiI6MTcyMDMzMzgwMSwicGF0aCI6Ii8zODM4MjYwNy8zNDYzMTU2MzQtMDBlOWI2MTMtZTJhZS00ODZiLTlhNDMtMzE1OTkzYWNkOGEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDA2MzAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjNzgzZjk3OWM0NzZmNjRhZjBiZTJiMTc3MTM4YmI2NzJmMWNmMzA1NTJlNDUwMmVhYmE2ZTQ3NTNkNzhmZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sKdMY51Fax9i1NlVQWDfz7nHrH1r7b-gSX-6h0Q37J8)
> 
> But with Citizen: ![image](https://private-user-images.githubusercontent.com/38382607/346315666-e2c1c27d-aad9-4114-9611-1d8a5941384b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzMzQxMDEsIm5iZiI6MTcyMDMzMzgwMSwicGF0aCI6Ii8zODM4MjYwNy8zNDYzMTU2NjYtZTJjMWMyN2QtYWFkOS00MTE0LTk2MTEtMWQ4YTU5NDEzODRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDA2MzAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjMmMzYTZkZThlMGY2M2NlMWFjYzMxZjI4MjBjOWExOTMyNDBjZmNmZTQ1MWE2Zjg2YjYzYzc2NmQ4NGJmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6jRsF2wOYmfcP3DW1yj87tJjVUYd29xDmkGFeNS_IDQ)

This should be patched now.
